source/adios2/helper/adiosXML.cpp

```diff
@@ -11,473 +11,162 @@
-                " can't have attributes, in call to ADIOS constructor\n");
-        }
-    }
-    else // opening tag
-    {
-        attributes = lf_GetAttributes(openingTag);
-    }
-    return attributes;
-}
-
-void InitXML(const std::string configXML, const MPI_Comm mpiComm,
-             const bool debugMode,
-             std::vector<std::shared_ptr<Transform>> &transforms,
-             std::map<std::string, IO> &ios)
-{
-    // independent IO
-    std::string fileContents(FileToString(configXML));
-    if (fileContents.empty())
-    {
-        // issue a warning?
-        return;
-    }
-
-    RemoveCommentsXML(fileContents);
-
-    // adios-config
-    std::string::size_type position(0);
-    const TagXML adiosConfigXML(
-        GetTagXML("adios-config", fileContents, position));
-
-    // process transforms, not yet implemented
-
-    while (position != std::string::npos)
-    {
-        const TagXML transformXML(
-            GetTagXML("transform ", adiosConfigXML.Elements, position));
-
-        if (transformXML.Header.empty())
-        {
-            break;
+            continue;
         }
-        // InitTransform(transformTag, debugMode, transforms);
-    }
-
-    position = 0;
-    // process IOs
-    while (position != std::string::npos)
-    {
-        // io
-        const TagXML ioXML(GetTagXML("io ", adiosConfigXML.Elements, position));
 
-        if (ioXML.Header.empty()) // no more groups to find
-        {
-            break;
-        }
-        InitIOXML(ioXML, mpiComm, debugMode, transforms, ios);
+        params.emplace(attrKey.value(), attrValue.value());
     }
+    return params;
 }
 
-void InitIOXML(const TagXML &ioXML, const MPI_Comm mpiComm,
+void InitIOXML(pugi::xml_node ioNode, const MPI_Comm mpiComm,
                const bool debugMode,
                std::vector<std::shared_ptr<Transform>> &transforms,
                std::map<std::string, IO> &ios)
 {
-    const Params ioAttributes(GetTagAttributesXML(ioXML.Header));
-
-    std::string ioName;
-    for (const auto &ioAttribute : ioAttributes)
+    // Extract <io name=""> attribute
+    pugi::xml_attribute nameAttr = ioNode.attribute("name");
+    if (!nameAttr)
     {
-        if (ioAttribute.first == "name")
-        {
-            ioName = ioAttribute.second;
-        }
-    }
-
-    if (debugMode)
-    {
-        if (ioName.empty())
-        {
-            throw std::invalid_argument(
-                "ERROR: io name=\"value\" attribute not found in opening XML "
-                "tag " +
-                ioXML.Header +
-                ", check XML config file, in call to ADIOS constructor\n");
-        }
-
-        if (ios.count(ioName) == 1) // io exists
+        if (debugMode)
         {
-            throw std::invalid_argument("ERROR: io name " + ioName +
-                                        " must be unique in XML config file, "
-                                        "in call to ADIOS constructor\n");
```

Comment:
InitXML has no idea where it's being called from.  Really none of the helper functions do since they're stand-alone and not members.  Now, if the parsing was in a private member function of the ADIOS class, which would make sense if nothing else is supposed to call to call it, then it would be feasible to do say the error was coming from the ADIOS class.  As a stand alone non-member function, it's an inversion of ownership though.  If the parsing moves to the ADIOS class then I'll support changing the messages but that's outside the scope of this PR.